gradle-plugin/build.gradle

```diff
@@ -23,18 +23,43 @@ gradlePlugin {
   }
 }
 
-pluginBundle {
-  website = 'https://pekko.apache.org/docs/pekko-grpc/current/'
-  vcsUrl = 'https://github.com/apache/incubator-pekko-grpc'
-  description = 'Building streaming gRPC servers and clients on top of Apache Pekko Streams'
-  tags = ['pekko', 'streams', 'reactive']
+publishing {
+    publications.withType(MavenPublication) {
+        pom {
+            name = "pekko-grpc-gradle-plugin"
+            description = "Apache Pekko gRPC - Support for building streaming gRPC servers and clients on top of Pekko Streams."
+            url = "https://pekko.apache.org/"
+            licenses {
+                license {
+                    name = "Apache-2.0"
+                    url = "https://www.apache.org/licenses/LICENSE-2.0.html"
+                }
+            }
+            developers {
+                developer {
+                    id = 'contributors'
+                    name = 'Contributors'
+                    url = 'https://github.com/apache/incubator-pekko-grpc/graphs/contributors'
+                    email = 'dev@pekko.apache.org'
+                }
+            }
+            scm {
+                connection = 'git@github.com:apache/incubator-pekko-grpc'
+                url = 'https://github.com/apache/incubator-pekko-grpc'
+            }
+        }
+    }
+}
 
-  plugins {
-    pekkoGrpcPlugin {
-      id = 'org.apache.pekko.grpc.gradle'
-      displayName = 'Apache Pekko gRPC'
+nexusPublishing {
+    repositories {
+        sonatype {
+            nexusUrl.set(uri("https://repository.apache.org/"))
+            snapshotRepositoryUrl.set(uri("https://repository.apache.org/content/repositories/snapshots/"))
```

Comment:
so when we need a release manager (a human) to do a release, they will need to hand edit this?

Reply:
the release URL will be something like https://repository.apache.org/service/local/staging - I need to track the correct path but it is something similar to this.

Reply:
> so when we need a release manager (a human) to do a release, they will need to hand edit this?

Nope, aside from maybe the version (but we have the same issue with sbt)

> the release URL will be something like https://repository.apache.org/service/local/staging - I need to track the correct path but it is something similar to this.

This should already be handled by the various `publishToSonatype` tasks and its variants as described in https://github.com/gradle-nexus/publish-plugin#configure-signing. The `https://repository.apache.org/service/local/staging` uri is nothing that should ever be set, thats an internal detail of how bundle/standard publishing is done by the plugin. If you check sbt-apache-sonatype (https://github.com/mdedetrich/sbt-apache-sonatype/blob/main/src/main/scala/org/mdedetrich/apache/sonatype/ApacheSonatypePlugin.scala#L36) you can see we only set the URI in on place.

In summary its the role of the plugins (i.e. gradle nexus publish-plugin/sbt-sonatype) to configure the staging repos, you only need to set the base URI. What may be a difference with the gradle nexus publish-plugin is that it may not support bundle publishing, but there isn't much we can do about that and its also not that critical since its just a single artifact and not many.

Reply:
you've hardcoded a snapshot url into this publish section so I do not see how this publish can handle when we need to do a non-snapshot release - we are actually waiting to do a release on this very repo

Reply:
This is how its documented to be done, from https://github.com/gradle-nexus/publish-plugin#publishing-to-maven-central-via-sonatype-ossrh

> Important. Users registered in Sonatype after [24 February 2021](https://central.sonatype.org/news/20210223_new-users-on-s01/) need to customize the following URLs:
>
 ```
  nexusPublishing {
    repositories {
        sonatype {  //only for users registered in Sonatype after 24 Feb 2021
            nexusUrl.set(uri("https://s01.oss.sonatype.org/service/local/"))
            snapshotRepositoryUrl.set(uri("https://s01.oss.sonatype.org/content/repositories/snapshots/"))
        }
     }
  }
  ```

Reply:
I understand now. Thanks.